core/components/atoms/selectionCard/SelectionCard.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import classNames from 'classnames';
+import { BaseHtmlProps, BaseProps } from '@/utils/types';
+import { useMultiSelect, useSingleSelect } from './hooks';
+
+type ClickEventType = React.MouseEvent<HTMLDivElement> | React.KeyboardEvent;
+
+export interface SelectionCardProps extends BaseProps, BaseHtmlProps<HTMLDivElement> {
+  /**
+   * Element to be rendered inside card item
+   */
+  children: React.ReactNode;
+  /**
+   * Provide unique name to each card when used in group
+   */
+  name?: string;
+  /**
+   * Disable the Selection Card
+   */
+  disabled?: boolean;
+  /**
+   * Handler to be called when Selection card is clicked
+   */
+  onClick?: (e: ClickEventType, name?: string) => void;
+  /**
+   * Describe z-index for overlay
+   */
+  zIndex?: number;
+  /**
+   * Defines if card is selected
+   */
+  selected?: boolean;
+}
+
+export const SelectionCard = (props: SelectionCardProps) => {
+  const { children, onClick, disabled, name, zIndex, selected, className, ...rest } = props;
+
+  const classes = classNames(
+    {
+      ['Selection-card']: true,
+      ['Selection-card--selected']: selected,
+      ['Selection-card--disabled']: disabled && !selected,
+      ['Selection-card--selected-disabled']: disabled && selected,
+    },
+    className
+  );
+
+  const onClickHandler = (e: ClickEventType) => {
+    onClick && onClick(e, name);
+  };
+
+  const onKeyDownHandler = (event: React.KeyboardEvent) => {
+    if (event.key === 'Enter') {
+      onClickHandler(event);
+    }
+  };
+
+  return (
+    <div
+      role="button"
```

Comment:
Please use "checkbox" role if possible, as it has single selection and multi-selection out of given options.